### PR TITLE
add method to generate unique token and reply with link

### PR DESCRIPTION
**Pull request by RitikJaiswal75**
This PR generates a link to my site with a unique token for the user who sends `/verify`.
This Part just consists of generating a token. The token is not saved to the RDS backend yet.
Will add the part saving data in the RDS backend in future PRs.

To test out the PR in action please join my server: https://discord.gg/tRQrgm7D
and send the `/hello` and `/verify` commands.

### CODE REVIEW

File: src/controllers/verifyCommand.ts [hunk 1]
```diff
@@ -1,8 +1,12 @@
...
 import { env } from "../typeDefinitions/default.types";
 import { discordTextResponse } from "../utils/discordResponse";
+import { generateUniqueToken } from "../utils/generateUniqueToken";
 import { sendDiscordDm } from "../utils/sendDiscordDm";
 
 export async function verifyCommand(userId: number, env: env) {
-  await sendDiscordDm(userId, env);
+  const token = await generateUniqueToken();
+  const url = `${VERIFICATION_SITE}/?token=${token}`;
+  await sendDiscordDm(userId, env, url);
```
Comment (rohan09-raj) [inline src/controllers/verifyCommand.ts:10]: Here we are passing a URL but in the sendDiscordDm util, the parameter is a token. Is this correct or am I missing something here?
image

File: src/utils/generateUniqueToken.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+export const generateUniqueToken = async () => {
+  const uuidToken = crypto.randomUUID();
+  const randomNumber = Math.floor(Math.random() * 1000000);
```
Comment (rohan09-raj) [inline src/utils/generateUniqueToken.ts:3]: Can we please store large numbers in a constant?

**Review by rohan09-raj — COMMENTED** [review]
The PR looks great, adding some comments and observations that I noticed.
- Should the `package-lock.json` be changed as there are no package upgrades, additions or removals ?
- Also, I noticed one thing that even when we get `Application did not respond` we get the DM, that is actually a weird behaviour and can confuse the user.

File: src/utils/generateUniqueToken.ts (hunk 2 above)
Comment (RitikJaiswal75) [reply src/utils/generateUniqueToken.ts:3]: isn't it already a constant

File: src/controllers/verifyCommand.ts (hunk 1 above)
Comment (RitikJaiswal75) [reply src/controllers/verifyCommand.ts:10]: sorry i missed that

File: src/utils/generateUniqueToken.ts (hunk 2 above)
Comment (rohan09-raj) [reply src/utils/generateUniqueToken.ts:3]: I am talking about the `1000000`

Comment (RitikJaiswal75) [reply src/utils/generateUniqueToken.ts:3]: The application did not respond only comes when the application is just now deployed. if you try on my bot now it won't show app did not respond

also for the package lock, I installed some packages and then removed them as they were not required. If you try to see the changes you can see there is a change in integrity as a package is upgraded. That will cause package-lock.json to change. it did not change package.json as the package updated was a dependency/funding for a working package.

As for the constant, just for 1 number, and that too is used in a single place I don't think it requires a separate constant and import.

Comment (yesyash) [reply src/utils/generateUniqueToken.ts:3]: Do we also verify the generated token at any place? 
If so, how will we do it as the encoded string (at line 6) has a random value i.e `randomNumber?

**Review by yesyash — DISMISSED** [review]

**Review by Maheima — APPROVED** [review]
LGTM!

File: src/utils/generateUniqueToken.ts (hunk 2 above)
Comment (harshith-venkatesh) [reply src/utils/generateUniqueToken.ts:3]: That's a great question

**Comment by RitikJaiswal75** [comment]
hi @yesyash We can not verify something completely random. To test it out we can store the random token and then check if the new token generated is different.

File: src/utils/generateUniqueToken.ts (hunk 2 above)
Comment (RitikJaiswal75) [reply src/utils/generateUniqueToken.ts:3]: hi @yesyash We can not verify something completely random. To test it out we can store the random token and then check if the new token generated is different.

File: src/controllers/verifyCommand.ts [hunk 3]
```diff
@@ -1,8 +1,12 @@
...
 import { env } from "../typeDefinitions/default.types";
 import { discordTextResponse } from "../utils/discordResponse";
+import { generateUniqueToken } from "../utils/generateUniqueToken";
 import { sendDiscordDm } from "../utils/sendDiscordDm";
 
 export async function verifyCommand(userId: number, env: env) {
-  await sendDiscordDm(userId, env);
+  const token = await generateUniqueToken();
+  const url = `${VERIFICATION_SITE_URL}/?token=${token}`;
+  await sendDiscordDm(userId, env, url);
```
Comment (heyrandhir) [inline src/controllers/verifyCommand.ts:10]: The third parameter for `sendDiscordDm` currently is message. Can you please re verify.

**Review by heyrandhir — CHANGES_REQUESTED** [review]

File: src/utils/generateUniqueToken.ts (hunk 2 above)
Comment (yesyash) [reply src/utils/generateUniqueToken.ts:3]: That does not answer my first question are we verifying the generated token at any place ?

File: src/controllers/verifyCommand.ts (hunk 3 above)
Comment (RitikJaiswal75) [reply src/controllers/verifyCommand.ts:10]: yes

File: src/utils/generateUniqueToken.ts (hunk 2 above)
Comment (RitikJaiswal75) [reply src/utils/generateUniqueToken.ts:3]: yes it will be verified in my site

**Review by rohan09-raj — APPROVED** [review]

**Review by samyakshah3008 — APPROVED** [review]
Great work, especially I loved the function of generating unique token. 
Looks good to merge ✅

**Review by Maheima — APPROVED** [review]
LGTM!

**Review by heyrandhir — APPROVED** [review]
LGTM

**Review by harshith-venkatesh — APPROVED** [review]

**Pull request merged**